index.js

```diff
@@ -6,14 +6,18 @@ module.exports = x => {
 		x = x.slice();
 	}
 
+	const iterate = x[Symbol.iterator] && typeof x !== 'string' && !Buffer.isBuffer(x);
```

Comment:
`iterate` => `shouldIterate`

Maybe also add a comment on why we're special-casing string/Buffer, regarding perf.